packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -122,12 +137,247 @@ describe("<DateRangeInput>", () => {
+                    input.simulate("blur");
+                    expect(onChange.called).to.be.false;
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("removes error message if input is changed to an in-range date again", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    changeInputText(input, inputString);
+                    input.simulate("blur");
+
+                    const IN_RANGE_DATE_STR = START_STR;
+                    input.simulate("focus");
+                    changeInputText(input, IN_RANGE_DATE_STR);
+                    input.simulate("blur");
+                    assertInputTextEquals(input, IN_RANGE_DATE_STR);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            type OutOfRangeTestFunction = (input: WrappedComponentInput,
+                                           inputString: string,
+                                           boundary?: DateRangeBoundary) => void;
+
+            function _runTestForEachScenario(runTestFn: OutOfRangeTestFunction) {
+                const { START, END } = DateRangeBoundary; // deconstruct to keep line lengths under threshold
+                it("if start < minDate", () => runTestFn(getStartInput(root), OUT_OF_RANGE_START_STR, START));
+                it("if start > maxDate", () => runTestFn(getStartInput(root), OUT_OF_RANGE_END_STR, START));
+                it("if end < minDate", () => runTestFn(getEndInput(root), OUT_OF_RANGE_START_STR, END));
+                it("if end > maxDate", () => runTestFn(getEndInput(root), OUT_OF_RANGE_END_STR, END));
+            }
         });
 
-        it.skip("Typing an invalid date displays the error message and calls onError with Date(undefined)", () => {
-            expect(true).to.be.false;
+        describe("Typing an invalid date...", () => {
+
+            let onChange: Sinon.SinonSpy;
+            let onError: Sinon.SinonSpy;
+            let root: WrappedComponentRoot;
+
+            beforeEach(() => {
+                onChange = sinon.spy();
+                onError = sinon.spy();
+
+                const result = wrap(<DateRangeInput
+                    defaultValue={DATE_RANGE}
+                    invalidDateMessage={INVALID_MESSAGE}
+                    onError={onError}
+                />);
+                root = result.root;
+
+                // clear the fields *before* setting up an onChange callback to
+                // keep onChange.callCount at 0 before tests run
+                changeStartInputText(root, "");
+                changeEndInputText(root, "");
+                root.setProps({ onChange });
+
+                return { root, onError };
+            });
+
+            afterEach(() => {
+                onChange = null;
+                onError = null;
+                root = null;
```

Comment:
no need to reset to null after each test, as the beforeEach will reset them. should be safe to remove this `afterEach` entirely.